xdsl/dialects/vector.py

```diff
@@ -292,6 +302,78 @@ def get(mask_operands: list[Operation | SSAValue]) -> CreatemaskOp:
         )
 
 
+@irdl_op_definition
+class TransferReadOp(IRDLOperation):
+    name = "vector.transfer_read"
+
+    source = operand_def(TensorOrMemrefOf(Attribute))
+    indices = var_operand_def(IndexType)
+    padding = operand_def(Attribute)
+    mask = opt_operand_def(VectorType[I1])
+
+    permutation_map = prop_def(AffineMapAttr)
+    in_bounds = opt_prop_def(ArrayAttr[BoolAttr])
+
+    result = result_def(VectorType)
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True)]
+
+    def verify_(self):
+        assert isa(self.source.type, MemRefType[Attribute] | TensorType[Attribute])
+        assert isa(self.result.type, VectorType[Attribute])
+        # TODO verify.
+
+    @staticmethod
+    def get(
```

Comment:
Same as @superlopuh commented in your other PR (#3649), we have moved to using `__init__`